File: Java/src/test/java/com/gildedrose/GildedRoseTest.java
package com.gildedrose;

import static org.junit.jupiter.api.Assertions.assertEquals;

import com.gildedrose.categories.*;
import org.junit.jupiter.api.Test;

class GildedRoseTest {

    @Test
    void conjuredManaCakeTest() {
        Item[] items = new Item[] { new Item("Conjured Mana Cake", 5, 10) };
        GildedRose app = new GildedRose(
            items,
            new DefaultCategory(),
            new ConjuredManaCake()
        );
        app.updateQuality();
        Item item = app.getItems()[0];
        assertEquals("Conjured Mana Cake", item.name);
        assertEquals(4, item.sellIn);
        assertEquals(8, item.quality);
    }
}

File: csharpcore/GildedRoseTest.cs
using System.Collections.Generic;
using csharpcore.Categories;
using Xunit;

namespace csharpcore
{
    public class GildedRoseTest
    {
        [Fact]
        public void foo()
        {
            IList<Item> Items = new List<Item> { new Item { Name = "foo", SellIn = 0, Quality = 0 } };
            GildedRose app = new GildedRose(Items, new DefaultCategory());
            app.UpdateQuality();
            Assert.Equal("fixme", Items[0].Name);
        }
    }
}